cl/alerts/tasks.py

```diff
@@ -31,28 +32,80 @@ def make_alert_messages(
     d: Docket,
     new_des: QuerySet,
     email_addresses: "ValuesQuerySet[User, Optional[str]]",  # type: ignore
+    recap_email_recipients: list[str] = [],
 ) -> List[EmailMultiAlternatives]:
     """Make docket alert messages that can be sent to users
 
     :param d: The docket to work on
     :param new_des: The new docket entries
     :param email_addresses: A list of user email addresses to send to
+    :param recap_email_recipients: The recap_email addresses if needed to send
+    the first case-user notification email.
     :return: A list of email messages to send
     """
+
     case_name = trunc(best_case_name(d), 100, ellipsis="...")
-    subject_template = loader.get_template("docket_alert_subject.txt")
-    subject = subject_template.render(
-        {
-            "docket": d,
-            "count": new_des.count(),
-            "case_name": case_name,
-        }
-    ).strip()  # Remove newlines that editors can insist on adding.
-    email_context = {"new_des": new_des, "docket": d}
+    messages = []
     txt_template = loader.get_template("docket_alert_email.txt")
     html_template = loader.get_template("docket_alert_email.html")
-    messages = []
-    for email_address in email_addresses:
+    subject_template = loader.get_template("docket_alert_subject.txt")
+    subject_context = {
+        "docket": d,
+        "count": new_des.count(),
+        "case_name": case_name,
+        "first_email": False,
+        "auto_subscribe": False,
+    }
+    email_context = {
+        "new_des": new_des,
+        "docket": d,
+        "docket_alert": None,
+        "first_email": False,
+        "auto_subscribe": False,
+    }
+
+    # Concatenate current subscriptions addresses and recap_email_recipients
+    email_addresses_list = list(email_addresses) + recap_email_recipients
+    for email_address in email_addresses_list:
+        if email_address in recap_email_recipients:
+            user_profile = UserProfile.objects.get(recap_email=email_address)
+            # Override user's @recap.email with the actual user's email address
+            email_address = user_profile.user.email
+            docket_alert_exist = DocketAlert.objects.filter(
+                docket_id=d, user_id=user_profile.user.pk
```

Comment:
Shouldn't this be `docket=d`? I'm surprised a test isn't failing due to this mistake?

Reply:
Yeah, I changed it to `docket=d` however it seems that it was working well, looks like it's allowed to assign the docket like `docket_id=d`, and also confirm that tests are covering this code, if docket_alert_exist filter fails many tests crash.